Microsoft.Azure.Cosmos/src/Query/Core/ExecutionContext/CosmosQueryExecutionContextWithNameCacheStaleRetry.cs

```
@@ -5,16 +5,19 @@
 namespace Microsoft.Azure.Cosmos.Query.Core.ExecutionContext
 {
     using System;
+    using System.Collections.Generic;
+    using System.Linq;
     using System.Threading;
     using System.Threading.Tasks;
+    using Microsoft.Azure.Cosmos.Diagnostics;
     using Microsoft.Azure.Cosmos.Query.Core.QueryClient;
 
     internal sealed class CosmosQueryExecutionContextWithNameCacheStaleRetry : CosmosQueryExecutionContext
     {
         private readonly CosmosQueryContext cosmosQueryContext;
         private readonly Func<CosmosQueryExecutionContext> cosmosQueryExecutionContextFactory;
         private CosmosQueryExecutionContext currentCosmosQueryExecutionContext;
-        private bool alreadyRetried;
+        private bool isInitialExecution = true;
```

why are you flipping this? The value was set to default(T) before and now you are setting a value outside the constructor.

---

The name cache stale should only be handled for the first operation. If it's not the initial operation then it should throw. If a customer gets a result from container(RID 1), deletes and recreates container(RID 2)  then attempts to get another result from the initial query pointing to RID 1 then query should throw an exception. There is no way to get a valid result in this scenario.